### PR TITLE
Add documentation. #trivial

### PR DESCRIPTION
**Pull request by scottrhoyt**
This adds some documentation around `Endpoint`, `Image`, `Response`, and `MultiTarget`.

### CODE REVIEW

File: Sources/Moya/Response.swift [hunk 1]
```diff
@@ -38,14 +42,17 @@ public extension Response {
         return self
     }
 
+    /// Filters out responses of a specific status code, generating errors when others are encountered.
```
Comment (BasThomas) [inline Sources/Moya/Response.swift]: These just return `an Error`, right? Not multiple. Might want to make that more clear.

Comment (BasThomas) [inline Sources/Moya/Response.swift]: And maybe make it "Filters out the responses of the specified status code [..]"

Comment (scottrhoyt) [inline Sources/Moya/Response.swift]: Yeah, I was just going off of the preexisting documentation for consistency. But I will change that as well.

**Comment by BasThomas** [comment]
We might want to unify the header docs at one point (only `///` or your approach with `\**`), but lets leave that for another time.

**Comment by scottrhoyt** [comment]
Thus far my approach is to use `///` for single line documentation comments and `/**` for multiline documentation comments, but I agree we should make sure it is consistent. I was having trouble getting the markdown (e.g. `- parameters:` to correctly register on multiline with `///`.

**Comment by BasThomas** [comment]
You might have to use `- parameter name: name does thing` per parameter with `///`.

**Comment by scottrhoyt** [comment]
Ah, that might be it then. In that case, I think:

```swift
/**
 function

 - parameters:
    - x: Is an x
    - y: Is a y
*/
```

might be a touch more human readable than

```swift
/// function
///
/// parameter x: Is an x
/// parameter y: Is a y
```

But that's just personal preference. I could go either way.

**Comment by BasThomas** [comment]
I agree, but I think I like `///` above `/**`. 😛

**Pull request merged**